packages/front-end/components/Holdout/HoldoutSelect.tsx

```diff
@@ -66,22 +66,39 @@ export const HoldoutSelect = ({
     });
   }, [holdouts, experimentsMap, selectedProject, getDatasourceById]);
 
+  const [userSelectedNone, setUserSelectedNone] = useState(false);
+  const shouldReCheckHoldout = useRef(true);
```

Comment:
A ref and useEffect is a weird way to solve this problem and super twisty. I recommend taking a step back and doing this differently